Push docker images to docker hub & automate git tag/release

jiyeonmolloy opened the pull request:
Fixes #10

jiyeonmolloy committed:
Update docker workflow

will-molloy committed:
Update readme

will-molloy commented on .github/workflows/docker.yml:
Don't want build on PR anymore :)

will-molloy committed:
`--password-stdin`

will-molloy committed:
`docker/login-action`

will-molloy committed:
Remove gradle version

Not needed - released as docker image.

will-molloy committed:
Set version for tag/docker

will-molloy committed:
Push git tag

will-molloy committed:
Manual git tag command

That github action broken and not maintained
https://github.com/actions-ecosystem/action-push-tag/issues/10

will-molloy committed:
Try set GITHUB_TOKEN

will-molloy committed:
Try `softprops/action-gh-release`

will-molloy committed:
Set `permissions`

will-molloy committed:
generate_release_notes

will-molloy committed:
bump ver

will-molloy committed:
Remove PR build

will-molloy committed:
Tweak step names
